### PR TITLE
chore: refactor docs

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
It refactors the docs to make them more usable/readable.

In the process, I needed to run the tests for one single machine, and for that reason this PR is adding a make goal for that.

## Why is it important?
Improve readability of the docs, and usability of the project, as it will allow running the tests for one single machine.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -39,6 +41,13 @@ run-group: ve
 	ANSIBLE_LIMIT=${GROUP} make run
 	vagrant destroy -f ${HOSTS}
 
+run-machine: ve
+	vagrant up ${MACHINE}
+	vagrant status ${MACHINE}
+	vagrant ssh-config ${MACHINE} > ssh_config
+	ANSIBLE_LIMIT=${MACHINE} make run
+	vagrant destroy -f ${MACHINE}
```
Comment (v1v) [inline Makefile:49]:
```suggestion
	vagrant up $(MACHINE)
	vagrant status $(MACHINE)
	vagrant ssh-config $(MACHINE) > ssh_config
	ANSIBLE_LIMIT=$(MACHINE) make run
	vagrant destroy -f $(MACHINE)
```

File: README.md [hunk 2]
```diff
@@ -19,71 +19,151 @@ ubuntu1804-64 | :white_check_mark:
...
+
+## Running the tests
+
+Once the target machines are ready, you need Ansible to run the tests. Because they could involve lots of VMs and commands executed over SSH, these tests are slow (currently 15 minutes in total).
+
+The commands that are available for running the tests are:
+
+- `make run`: it will run the tests for all the OSs in the support matrix.
+- `make run-group`: it will run the tests for all the OSs in a group of OSs. For each ansible group, it will start the vm, check the VM's status, configure SSH, run the ansible playbook, and finally will destroy the vm.
+- `make batch`: it will run the tests in batches. Since it processes each group sequentially, it uses less CPU and memory.
```
Comment (v1v) [inline README.md:78]: What do you think to add those details to the Make goals? And then provide a `help` make goal that provide those details on the fly?

**Review by v1v — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (v1v) [reply Makefile:49]: I think it works in both cases

Comment (mdelapenya) [reply Makefile:49]: I'll update them both, to be consistent 👍

File: requirements.txt [hunk 3]
```diff
@@ -1,4 +1,4 @@
-ansible==2.8.8
+ansible==2.9.1
```
Comment (v1v) [inline requirements.txt:1]: Can you downgrade and try again? It might be related?

Comment (mdelapenya) [reply requirements.txt]: Reverted here: https://github.com/elastic/beats-tester/pull/208/commits/6a21cb009ef87f9d6fa5a31c025423d4b5af9837

File: roles/test-beat/tasks/main.yml [hunk 4]
```diff
@@ -16,7 +16,6 @@
   include_vars: '{{ item }}'
   with_first_found:
   - files:
-    - 'vars/{{ ansible_distribution_id | lower }}.yml'
```
Comment (mdelapenya) [inline roles/test-beat/tasks/main.yml:19]: See https://github.com/ansible/ansible/issues/70772

**Review by cachedout — APPROVED** [review]

File: requirements.txt (hunk 3 above)
Comment (mdelapenya) [reply requirements.txt:1]: I kept the version as it was impossible to run the test suite on Mac with 2.8.8

I checked that the very same error than with 2.8.8 is happening with 2.9.1, see https://github.com/elastic/beats-tester/pull/208/files#r761364594 to understand why

**Comment by mdelapenya** [comment]
I'll keep this open, as I see it very valuable to improve the docs here.

On the other hand, comparing the Vagrant approach with the Ansible + AWS approach from the e2e-testing repo, I like the latter more

**Comment by cachedout** [comment]
> I'll keep this open, as I see it very valuable to improve the docs here.

beats-tester is going to be deprecated. I think this is safe to close.

**Pull request closed without merging**